### DISCUSSION TITLE
install under PHP8.1

### DISCUSSION THREAD

**Discussion by phoarau** [chillerlan/php-qrcode#219] [General]
I've installed PHP-QRCODE under a PHP8.1 plateform. 
All seems to work well but it seems that it is a "flange version".
Not all the features, like Imagick are active.
Is there any new about the full compatibility for new version of PHP?

**Comment by codemasher** [comment]
I don't know exactly what you mean by "Not all the features, like Imagick are active". Can you please elaborate because that seems like the extension is not installed in your PHP environment (it's not part of PHP and needs to be [installed manually](https://php-qrcode.readthedocs.io/en/main/Usage/Installation.html#imagemagick)).

**Reply by phoarau** [reply]
Thank for your answer,
I mean that, all the needed extension (php-imagick for example) are installtedn on my PHP server, but this feature is not active with my php-qrcode installation.
Here my phpinfo imagick section :
`<html><head></head><body><h2><a name="module_imagick">imagick</a></h2>


imagick module | enabled
-- | --
imagick module version | 3.6.0
imagick classes | Imagick, ImagickDraw, ImagickPixel, ImagickPixelIterator, ImagickKernel
Imagick compiled with ImageMagick version | ImageMagick 6.9.11-60 Q16 x86_64 2021-01-25 https://imagemagick.org
Imagick using ImageMagick library version | ImageMagick 6.9.11-60 Q16 x86_64 2021-01-25 https://imagemagick.org
ImageMagick copyright | (C) 1999-2021 ImageMagick Studio LLC
ImageMagick release date | 2021-01-25
ImageMagick number of supported formats: | 237
ImageMagick supported formats | 3FR,  3G2, 3GP, AAI, AI, APNG, ART, ARW, AVI, AVIF, AVS, BGR, BGRA, BGRO, BIE,  BMP, BMP2, BMP3, BRF, CAL, CALS, CANVAS, CAPTION, CIN, CIP, CLIP, CMYK,  CMYKA, CR2, CR3, CRW, CUR, CUT, DATA, DCM, DCR, DCX, DDS, DFONT, DNG,  DPX, DXT1, DXT5, EPDF, EPI, EPS, EPS2, EPS3, EPSF, EPSI, EPT, EPT2,  EPT3, ERF, FAX, FILE, FITS, FRACTAL, FTP, FTS, G3, G4, GIF, GIF87,  GRADIENT, GRAY, GRAYA, GROUP4, H, HALD, HDR, HEIC, HISTOGRAM, HRZ, HTM,  HTML, HTTP, HTTPS, ICB, ICO, ICON, IIQ, INFO, INLINE, IPL, ISOBRL,  ISOBRL6, J2C, J2K, JBG, JBIG, JNG, JNX, JP2, JPC, JPE, JPEG, JPG, JPM,  JPS, JPT, JSON, K25, KDC, LABEL, M2V, M4V, MAC, MAGICK, MAP, MASK, MAT,  MATTE, MEF, MIFF, MKV, MNG, MONO, MOV, MP4, MPC, MPG, MRW, MSL, MTV,  MVG, NEF, NRW, NULL, ORF, OTB, OTF, PAL, PALM, PAM, PATTERN, PBM, PCD,  PCDS, PCL, PCT, PCX, PDB, PDF, PDFA, PEF, PES, PFA, PFB, PFM, PGM, PGX,  PICON, PICT, PIX, PJPEG, PLASMA, PNG, PNG00, PNG24, PNG32, PNG48, PNG64,  PNG8, PNM, POCKETMOD, PPM, PREVIEW, PS, PS2, PS3, PSB, PSD, PTIF, PWP,  RADIAL-GRADIENT, RAF, RAS, RAW, RGB, RGBA, RGBO, RGF, RLA, RLE, RMF,  RW2, SCR, SCT, SFW, SGI, SHTML, SIX, SIXEL, SPARSE-COLOR, SR2, SRF,  STEGANO, SUN, TEXT, TGA, THUMBNAIL, TIFF, TIFF64, TILE, TIM, TTC, TTF,  TXT, UBRL, UBRL6, UIL, UYVY, VDA, VICAR, VID, VIDEO, VIFF, VIPS, VST,  WBMP, WEBM, WEBP, WMV, WPG, X, X3F, XBM, XC, XCF, XPM, XPS, XV, XWD,  YCbCr, YCbCrA, YUV

</body></html>`

and here my cillerlan/php-qrcode/ssrc/Output/QROutputInterface.php 

`<?php
/**
 * Interface QROutputInterface,
 *
 * @filesource   QROutputInterface.php
 * @created      02.12.2015
 * @package      chillerlan\QRCode\Output
 * @author       Smiley <smiley@chillerlan.net>
 * @copyright    2015 Smiley
 * @license      MIT
 */

namespace chillerlan\QRCode\Output;

use chillerlan\QRCode\Data\QRMatrix;

/**
 * Converts the data matrix into readable output
 */
interface QROutputInterface{

	const DEFAULT_MODULE_VALUES = [
		// light
		QRMatrix::M_NULL            => false, // 0
		QRMatrix::M_DATA            => false, // 4
		QRMatrix::M_FINDER          => false, // 6
		QRMatrix::M_SEPARATOR       => false, // 8
		QRMatrix::M_ALIGNMENT       => false, // 10
		QRMatrix::M_TIMING          => false, // 12
		QRMatrix::M_FORMAT          => false, // 14
		QRMatrix::M_VERSION         => false, // 16
		QRMatrix::M_QUIETZONE       => false, // 18
		QRMatrix::M_LOGO            => false, // 20
		QRMatrix::M_TEST            => false, // 255
		// dark
		QRMatrix::M_DARKMODULE << 8 => true,  // 512
		QRMatrix::M_DATA << 8       => true,  // 1024
		QRMatrix::M_FINDER << 8     => true,  // 1536
		QRMatrix::M_ALIGNMENT << 8  => true,  // 2560
		QRMatrix::M_TIMING << 8     => true,  // 3072
		QRMatrix::M_FORMAT << 8     => true,  // 3584
		QRMatrix::M_VERSION << 8    => true,  // 4096
		QRMatrix::M_FINDER_DOT << 8 => true,  // 5632
		QRMatrix::M_TEST << 8       => true,  // 65280
	];

	/**
	 * generates the output, optionally dumps it to a file, and returns it
	 *
	 * @return mixed
	 */
	public function dump(string $file = null);

}
`

**Reply by codemasher** [reply]
The Imagick installation looks fine - does the php-qrcode imagemagick example ([v4.3.x](https://github.com/chillerlan/php-qrcode/blob/d5d728e79eeb6350632c2618f179b7d945e23f24/examples/imagick.php) or [v5.0-beta](https://github.com/chillerlan/php-qrcode/blob/ae0f31241554b509b62b54e7cd14ebd4d384ec16/examples/imagick.php)) run properly?

**Reply by phoarau** [reply]
Oh! the 4.3.x version is working well, but not the 5.0-beta one.
So may I get example from the older version and not the main branch?

**Reply by codemasher** [reply]
You can pick the version from [the branches list](https://github.com/chillerlan/php-qrcode/branches)! [Here, v4.3.x](https://github.com/chillerlan/php-qrcode/tree/v4.3.x/examples).
